tests/delete/tests.py

```diff
@@ -90,6 +90,30 @@ def test_protect_multiple(self):
         with self.assertRaisesMessage(ProtectedError, msg):
             a.protect.delete()
 
+    def test_protect_chained(self):
+        p = P.objects.create()
+        r = R.objects.create(p=p)
+        B.objects.create(protect=r)
+
+        msg = (
+            "Cannot delete some instances of model 'P' because they are "
+            "referenced through protected foreign keys: 'B.p'."
```

Comment:
Don't know the error message should be `R.p` or `B.p`?
If it's going to be `R.p` we can fix it by:

```diff
--- a/django/db/models/deletion.py
+++ b/django/db/models/deletion.py
@@ -297,7 +297,7 @@ class Collector:
                         field.remote_field.on_delete(self, field, sub_objs, self.using)
                     except ProtectedError as error:
                         key = "'%s.%s'" % (
-                            error.protected_objects[0].__class__.__name__,
+                            field.model.__name__,
                             field.name,
                         )
                         protected_objects[key] += error.protected_objects
```

@felixxm What do you think?

Reply:
Yes it should be `R.p`, we didn't take into account nested protected relations in the ab3cbd8b9a315911248227208630a020cedca08f (probably my fault). Also casting to list is not necessary anymore after this change.